src/Type/Accessory/AccessoryArrayListType.php

```diff
@@ -151,6 +151,13 @@ public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $uni
 			return $this;
 		}
 
+		if (
+			$valueType->isArray()->yes()
+			&& IntegerRangeType::fromInterval(0, null)->isSuperTypeOf($offsetType)->yes()
```

Comment:
I don't get this part at all.

1) Why does it matter what `$valueType` is?
2) List certainly isn't preserved if I assign any positive-int offset.